ext/storage_sources/azure_store/azure_storage_source.cpp

```diff
@@ -87,26 +91,85 @@ static int azure_file_read(WT_FILE_HANDLE *, WT_SESSION *, wt_off_t, size_t, voi
   __attribute__((__unused__));
 static int azure_file_size(WT_FILE_HANDLE *, WT_SESSION *, wt_off_t *) __attribute__((__unused__));
 
+// Return a customised file system to access the Azure storage source.
 static int
 azure_customize_file_system(WT_STORAGE_SOURCE *storage_source, WT_SESSION *session,
-  const char *bucket_name, const char *auth_token, const char *config,
-  WT_FILE_SYSTEM **file_systemp)
+  const char *bucket, const char *auth_token, const char *config, WT_FILE_SYSTEM **file_system)
 {
-    WT_UNUSED(storage_source);
-    WT_UNUSED(session);
-    WT_UNUSED(bucket_name);
-    WT_UNUSED(auth_token);
-    WT_UNUSED(config);
-    WT_UNUSED(file_systemp);
-
-    return 0;
+    if (bucket == nullptr || std::string(bucket).length() == 0) {
+        std::cerr << "azure_customize_file_system: Bucket not specified." << std::endl;
+        return EINVAL;
+    }
+
+    // Get any prefix to be used for the object keys.
+    WT_CONFIG_ITEM obj_prefix_config;
+    std::string obj_prefix;
+
+    // Get the value of the config key from the string
+    azure_store *azure = (azure_store *)storage_source;
+    int ret;
+    if ((ret = azure->wt_api->config_get_string(
+           azure->wt_api, session, config, "prefix", &obj_prefix_config)) == 0)
+        obj_prefix = std::string(obj_prefix_config.str, obj_prefix_config.len);
+    else if (ret != WT_NOTFOUND) {
+        std::cerr << "azure_customize_file_system: error parsing config for object prefix."
+                  << std::endl;
+        return ret;
+    }
+
+    // Fetch the native WT file system.
+    WT_FILE_SYSTEM *wt_file_system;
+    if ((ret = azure->wt_api->file_system_get(azure->wt_api, session, &wt_file_system)) != 0)
+        return ret;
+
+    // Get a copy of the home directory.
+    const std::string home_dir = session->connection->get_home(session->connection);
+
+    // Create the file system.
+    azure_file_system *azure_fs;
+    if ((azure_fs = (azure_file_system *)calloc(1, sizeof(azure_file_system))) == nullptr) {
+        std::cerr << "azure_customize_file_system: unable to allocate memory for file system."
+                  << std::endl;
+        return ENOMEM;
+    }
+
+    azure_fs->store = azure;
+    azure_fs->wt_fs = wt_file_system;
+    azure_fs->home_dir = home_dir;
+
+    try {
+        azure_fs->azure_conn = new azure_connection(bucket, obj_prefix);
+    } catch (std::runtime_error &e) {
+        std::cerr << std::string("azure_customize_file_system: ") + e.what() << std::endl;
+        return ENOENT;
```

Comment:
could you clarify why ENOENT is used here instead of EINVAL?

Reply:
Using ENOENT because "no such file or directory" fits better with error